ethereum-common/src/utils.rs

```diff
@@ -103,3 +112,29 @@ pub fn bitfield_bytes_tree_hash_root<const N: usize>(bytes: &[u8]) -> Hash256 {
         .finish()
         .expect("bitfield tree hash buffer should not exceed leaf limit")
 }
+
+pub fn rlp_encode_transaction_index(index: &u64) -> Vec<u8> {
+    let mut buf = Vec::with_capacity(100);
+    Encodable::encode(&index, &mut buf);
+
+    buf
+}
+
+pub fn rlp_encode_index_and_receipt(
+    index: &u64,
+    receipt: &ReceiptEnvelope<Log>,
+) -> (Vec<u8>, Vec<u8>) {
+    let mut buf = Vec::with_capacity(CAPACITY_RLP_RECEIPT);
+    receipt.encode_2718(&mut buf);
+
+    (rlp_encode_transaction_index(index), buf)
+}
+
+pub fn rlp_encode_receipts_and_nibble_tuples(receipts: &[Receipt]) -> Vec<(Vec<u8>, Vec<u8>)> {
+    receipts
+        .iter()
+        .map(|(transaction_index, receipt)| {
+            rlp_encode_index_and_receipt(transaction_index, receipt)
+        })
+        .collect::<Vec<_>>()
+}
```

Comment:
Why not in `rlp_node_codec`?

Reply:
there is not much sense for it since `rlp_node_codec` module provides specific implementation whereas these funcs are just utility

Reply:
Are these used directly by code that uses `ethereum-common`?

Reply:
yep. It is required by the funds relayer and the corresponding gear-program to verify patricia-proofs